app/src/main/cpp/skyline/services/account/IProfile.cpp

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include <kernel/types/KProcess.h>
+#include "IProfile.h"
+
+namespace skyline::service::account {
+    IProfile::IProfile(const DeviceState &state, ServiceManager &manager, const UserId &userId) : BaseService(state, manager, Service::account_IProfile, "account:IProfile", {
+        {0x0, SFUNC(IProfile::Get)}
+    }) {}
+
+    void IProfile::Get(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        AccountUserData userData{};
+        AccountProfileBase profileBase{};
+
+        std::string username = state.settings->GetString("username_value");
+        size_t usernameSize = std::min(sizeof(profileBase), username.size());
```

Comment:
Not sure this is correct, since the struct contains more than the username. The username max size can be checked through .size() if you make it an std::array. You could also have the ProfileBase constructor take a string.

Reply:
Oh, that's a typo. Should have been sizeof(profileBase.nickname)